src/acl/Certificate.h

```diff
@@ -17,7 +17,7 @@ namespace Acl
 {
 
 /// a "user_cert" or "ca_cert" ACL
-class ClientCertificateCheck: public ParameterizedNode< ACLData<X509 *> >
+class ClientCertificateCheck: public ParameterizedNode< ACLData<Security::Certificate *> >
```

Comment:
I have removed the "so that it can pass stricter validation checksSecurity" reason in the original PR description because, IMO, we should not be testing OpenSSL-specific code outside of OpenSSL builds (even if this file can be successfully tested today after this change). For example, I should be able to add an OpenSSL-specific field to ClientCertificateCheck _without_ breaking any tests!